core/network/src/service.rs

```diff
@@ -579,6 +583,9 @@ fn run_thread<B: BlockT + 'static, S: NetworkSpecialization<B>, H: ExHashT>(
 			}
 		}
 
+		is_offline.store(protocol.is_offline(), Ordering::Relaxed);
+		is_major_syncing.store(protocol.is_major_syncing(), Ordering::Relaxed);
```

Comment:
What's the rationale for moving those here? Previously they would only be updated when something changed, as opposed to at each call to `poll` as is done here. Also, since the logic is inside of `ChainSync`, why move it to here and expose it via those methods?

Reply:
I'm trying to unbloat `sync.rs` and `protocol.rs` by isolating code as much as possible from the threading strategy. To me `sync.rs` really should only ever be a state machine that decides who to send messages to, and shouldn't be concerned with updating the content of `Arc`s or knowing about channels.

If the boolean update is an issue, a solution would be to make methods from `ChainSync` return a value indicating that there was a state change in `is_offline` and/or `is_major_syncing`.

Reply:
To expand: to me the situation in `sync.rs` and `protocol.rs` is catastrophic to me. People (including me) don't want to work on this part of the code because it is so horribly bloated and complicated.
And we don't even have warp sync or anything like that. This is just the very basic straight-forward "let's pull blocks" strategy that really shouldn't be that hard.

Reply:
Ok, and how does moving those operations to here improves the situation? Whether it happens somewhere inside `ChainSync`, or here, with the current approach it will still run in the same task as a set of sequential steps(for which there is indeed something to be said).

`is_offline.store(protocol.is_offline(), Ordering::Relaxed);` is certainly simpler than the equivalent that was found in `ChainSync`, on the other hand I'm not sure if there was anything incorrect about the previous code.

Reply:
Well, simplification. I don't think there was anything correct indeed, but it clearly makes the code of `ChainSync` more complex and strongly discourages any improvement to it.

Reply:
Ok. The previous was more complicated, and it was also more targeted. However I don't think it's an important optimization...